icepyx/core/read.py

```diff
@@ -752,9 +817,16 @@ def load(self):
         # In these situations, xarray recommends manually controlling the merge/concat process yourself.
         # While unlikely to be a broad issue, I've heard of multiple matching timestamps causing issues for combining multiple IS2 datasets.
         for file in self.filelist:
+            if file.startswith("s3"):
+                # If path is an s3 path create an s3fs filesystem to reference the file
+                # TODO would it be better to be able to generate an s3fs session from the Mixin?
+                s3 = earthaccess.get_s3fs_session(daac="NSIDC", provider=self.auth)
```

Comment:
Good question. I searched and currently this is the second module (other than is2ref.py) that we use `get_s3fs_session()`. With three uses I'm fine to leave it here for now, but I think if we add another in the future the answer to your question would be a resounding "yes".